src/taoverse/model/model_updater.py

```diff
@@ -133,12 +139,17 @@ async def sync_model(
         self.model_tracker.on_miner_model_updated(hotkey, metadata)
 
         # Check that the hash of the downloaded content matches.
-        secure_hash = get_hash_of_two_strings(model.id.hash, hotkey)
-        if secure_hash != metadata.id.secure_hash:
-            raise MinerMisconfiguredError(
-                hotkey,
-                f"Hash of content downloaded from hugging face does not match chain metadata. {metadata}",
-            )
+        if model.id.hash != metadata.id.hash:
```

Comment:
Can you add a comment that we check this case first for legacy competition reasons that occurred before the hash was secure? And that in 'modern' competitions the metadata.id.hash is always None?

I just see this being a potentially confusing section of code otherwise.

Reply:
I added that. However, I need to deprecate the optional securing of the hash in model upload.